src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -474,16 +474,29 @@ export class IndexPatternsService {
   }
 
   /**
-   * Create a new index pattern and save it right away
+   * Create a new index pattern and save it right away and make the default
+   * index pattern.
+   *
    * @param spec
-   * @param override Overwrite if existing index pattern exists
-   * @param skipFetchFields
+   * @param override Overwrite if existing index pattern exists.
+   * @param skipFetchFields Whether to skip field refresh step.
+   * @param makeDefault Whether to make the new index pattern the default
+   *                    index pattern.
    */
 
-  async createAndSave(spec: IndexPatternSpec, override = false, skipFetchFields = false) {
+  async createAndSave(
+    spec: IndexPatternSpec,
+    override = false,
+    skipFetchFields = false,
+    makeDefault: boolean = true
+  ) {
```

Comment:
I don't think 'makeDefault' should default to true. Seems like a less common use case.

Reply:
> Is it necessary to add the `makeDefault` functionality directly to this function? I'm prefer to combine simpler functions.

I guess it is not, but the `this.setDefault(...` call was already part of this function. This change just adds ability to make that call optional.

> I don't think 'makeDefault' should default to true. Seems like a less common use case.

It was the only case before this change. I made it default to true to make a non-breaking change to this function signature. Because before this function would always make the index patter default, and by setting `makeDefault` to default to `true` it still preserves the old behaviour.

Are you OK to go ahead with this change? Or I should I remove `makeDefault` option from the HTTP API? I could also make the `makeDefault` default to `false` or move the `this.setDefault(...` call out of this function but would be a breaking change, it would affect all consumers of this function.

Reply:
>This change just adds ability to make that call optional.

`setDefault` really only does something if a default isn't already set or the second parameter - `force` - is set true. As best I can tell your change really only results in allowing opting out of setting the default index pattern. I don't think there's a use case for that.

Reply:
Then I'll remove the `makeDefault` altogether, didn't expect `setDefault` to have an optional `force` argument.